roles/metrics-export.lua

```diff
@@ -221,17 +231,27 @@ local function validate_http(conf)
     end
 
     for i, nodei in ipairs(conf) do
-        local hosti, porti, erri = parse_listen(nodei.listen)
+        local listen_address = nodei.listen
+        if listen_address == nil then
+            goto continue
+        end
+        local hosti, porti, erri = parse_listen(listen_address)
         assert(erri == nil) -- We should already successfully parse the URI.
         for j, nodej in ipairs(conf) do
             if i ~= j then
-                local hostj, portj, errj = parse_listen(nodej.listen)
+                listen_address = nodej.listen
+                if listen_address == nil then
+                    goto continue
+                end
+                local hostj, portj, errj = parse_listen(listen_address)
                 assert(errj == nil) -- The same.
                 if hosti == hostj and porti == portj then
                     error("http configuration nodes must have different listen targets", 2)
```

Comment:
I think we need the same check (for unique) for a `server`.

Reply:
Added logic for this validation.